test/generators/javascript/JavaScriptGenerator.spec.ts

```diff
@@ -6,6 +6,41 @@ describe('JavaScriptGenerator', () => {
     generator = new JavaScriptGenerator();
   });
 
+  test('should not render `class` with reserved keyword', async () => {
+    const doc = {
+      $id: 'Address',
+      type: 'object',
+      properties: {
+        enum: { type: 'string' },
+        reservedEnum: { type: 'string' }
+      },
+      additionalProperties: false
+    };
+    const expected = `class Address {
+  reservedReservedEnum;
+  reservedEnum;
+
+  constructor(input) {
+    this.reservedReservedEnum = input.reservedReservedEnum;
```

Comment:
`enum` isn't keyword in JS, also not `interface` or `type` :)

Reply:
Based on https://www.w3schools.com/js/js_reserved.asp they are?

Reply:
LOL https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#future_reserved_keywords I don't know about it. Ok, leave it :)